### ISSUE TITLE
nginx.conf failing after adding and removing nginx.conf.template

### ISSUE THREAD

**Issue by Corstiaan84** [dokku/dokku#2132]
Description of problem:
After adding a nginx.conf.template, deploys fail with:

````remote: nginx: [emerg] unknown log format "main" in /home/dokku/everseller/nginx.conf:117
remote: nginx: configuration file /etc/nginx/nginx.conf test failed````

After removing the nginx.conf.template, deploys still fail with same message.

Output of the following commands

- `uname -a`: Linux superbad 3.13.0-79-generic #123-Ubuntu SMP Fri Feb 19 14:27:58 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux

- `free -m`:
             total       used       free     shared    buffers     cached
Mem:          3953       3767        185         12        137       1273
-/+ buffers/cache:       2357       1596
Swap:         4092        442       3650

- `docker version`:
 Version:      1.10.3
 API version:  1.22
 Go version:   go1.5.3
 Git commit:   20f81dd
 Built:        Thu Mar 10 15:54:52 2016
 OS/Arch:      linux/amd64
Cannot connect to the Docker daemon. Is the docker daemon running on this host?

- `docker -D info`:
Cannot connect to the Docker daemon. Is the docker daemon running on this host?

- `sigil -v`:
sigil: command not found

- `docker run -ti gliderlabs/herokuish:latest herokuish version`:
docker: Cannot connect to the Docker daemon. Is the docker daemon running on this host?.

- `dokku version`:
0.4.14

- `dokku plugin`:
plugn: dev
  00_dokku-standard    0.4.0 enabled    dokku core standard plugin
  20_events            0.4.0 enabled    dokku core events logging plugin
  app-predeploy-tasks  0.1.0 enabled    Run arbitrary app tasks prior to dokku's deploy phase
  apps                 0.4.0 enabled    dokku core apps plugin
  backup               0.4.0 enabled    dokku core backup plugin
  build-env            0.4.0 enabled    dokku core build-env plugin
  certs                0.4.0 enabled    dokku core certificate management plugin
  checks               0.4.0 enabled    dokku core checks plugin
  common               0.4.0 enabled    dokku core common plugin
  config               0.4.0 enabled    dokku core config plugin
  docker-options       0.4.0 enabled    dokku core docker-options plugin
  domains              0.4.0 enabled    dokku core domains plugin
  enter                0.4.0 enabled    dokku core enter plugin
  git                  0.4.0 enabled    dokku core git plugin
  logs                 0.4.0 enabled    dokku core logs plugin
  mysql                1.0.0 enabled    dokku mysql service plugin
  named-containers     0.4.0.1 enabled    dokku core named containers plugin
  nginx-vhosts         0.4.0 enabled    dokku core nginx-vhosts plugin
  plugin               0.4.0 enabled    dokku core plugin plugin
  postgres             1.0.0 enabled    dokku postgres service plugin
  ps                   0.4.0 enabled    dokku core ps plugin
  shell                0.4.0 enabled    dokku core shell plugin
  tags                 0.4.0 enabled    dokku core tags plugin
  tar                  0.4.0 enabled    dokku core tar plugin

- `cat /home/dokku/<app>/nginx.conf` (if applicable):
cat: /home/dokku/everseller/nginx.conf: Permission denied

- Link to the exact repository being deployed (if possible/applicable):
- If a deploy is failing or behaving unexpectedly:
  - ruby/rails
  - If it exists, the contents of your `Procfile`.
web: bundle exec unicorn -p $PORT -c ./config/unicorn.rb
worker: bundle exec rake qc:work QUEUES="default,mailers"

Environment details (AWS, VirtualBox, physical, etc.):
VPS, trans-ip

How was dokku installed?:
apt-get

How reproducible:
Add a conf template, deploy. Remove template, deploy.

Actual Results:
See description

Expected Results:
A succesful deploy

**Comment by josegonzalez** [comment]
I believe we don't remove the `nginx.conf.template` from `/home/dokku/<app>/nginx.conf.template` under `0.4.14`, and fixed that in 0.5.x. Can you check to see if that file is there, remove it if it is, and `ps:rebuild` once that is done?

**Comment by Corstiaan84** [comment]
Awesome -- working :-)

This is my template. I am assuming many errors. Can you point them out for me?

````# Special characters - dollar signs, spaces inside of quotes, etc. -
# should be escaped with a single backslash or can cause deploy failures.

server {
  listen      [::]:80;
  listen      80;
  server_name $NOSSL_SERVER_NAME;
  access_log  /var/log/nginx/${APP}-access.log;
  error_log   /var/log/nginx/${APP}-error.log;

  # set a custom header for requests
  # add_header X-Served-By www-ec2-01;

  location    / {
    proxy_pass  http://$APP;
    proxy_http_version 1.1;
    proxy_set_header Upgrade \$http_upgrade;
    proxy_set_header Connection "upgrade";
    proxy_set_header Host \$http_host;
    proxy_set_header X-Forwarded-Proto \$scheme;
    proxy_set_header X-Forwarded-For \$remote_addr;
    proxy_set_header X-Forwarded-Port \$server_port;
    proxy_set_header X-Request-Start \$msec;
  }

  include $DOKKU_ROOT/$APP/nginx.conf.d/*.conf;

  # Proxy download 
  location ~* ^/internal_redirect/(.*?)/(.*) {
  # Do not allow people to mess with this location directly
  # Only internal redirects are allowed
  internal;

  # Location-specific logging
  access_log logs/internal_redirect.access.log main;
  error_log logs/internal_redirect.error.log warn;

  # Extract download url from the request
  set $download_uri \$2;
  set $download_host \$1;

  # Compose download url
  set $download_url http://\$download_host/\$download_uri;

  # Set download request headers
  proxy_set_header Host \$download_host;
  proxy_set_header Authorization '';

  # The next two lines could be used if your storage 
  # backend does not support Content-Disposition 
  # headers used to specify file name browsers use 
  # when save content to the disk
  proxy_hide_header Content-Disposition;
  add_header Content-Disposition 'attachment; filename="\$args"';

  # Do not touch local disks when proxying 
  # content to clients
  proxy_max_temp_file_size 0;

  # Download the file and send it to client
  proxy_pass \$download_url;
}
}
````
